codec-http2/src/main/java/io/netty/handler/codec/http2/DelegatingDecompressorFrameListener.java

```diff
@@ -38,7 +38,10 @@
  * stream. The decompression provided by this class will be applied to the data for the entire stream.
  */
 public class DelegatingDecompressorFrameListener extends Http2FrameListenerDecorator {
-    private static final Http2ConnectionAdapter CLEAN_UP_LISTENER = new Http2ConnectionAdapter() {
+
+    private final Http2Connection connection;
+    private final boolean strict;
+    private final Http2ConnectionAdapter cleanUpListener = new Http2ConnectionAdapter() {
```

Comment:
Maybe we should delay the initialization of this to after `propertyKey` is initialized because it depends on it?